packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/frescosupport/FrescoBasedReactTextInlineImageSpan.kt

```diff
@@ -0,0 +1,172 @@
+
+    override val width: Int
+        get() = _width
+
+    override val height: Int
+        get() = _height
+
+    override var drawable: Drawable? = null
+        private set
+
+    /**
+     * The ReactTextView that holds this ImageSpan is responsible for passing these methods on so
+     * that we can do proper lifetime management for Fresco
+     */
+    public override fun onDetachedFromWindow() {
+        draweeHolder.onDetach()
+    }
+
+    public override fun onStartTemporaryDetach() {
+        draweeHolder.onDetach()
+    }
+
+    public override fun onAttachedToWindow() {
+        draweeHolder.onAttach()
+    }
+
+    public override fun onFinishTemporaryDetach() {
+        draweeHolder.onAttach()
+    }
+
+    public override fun getSize(
+        paint: Paint,
+        text: CharSequence,
+        start: Int,
+        end: Int,
+        fm: Paint.FontMetricsInt?
+    ): Int {
+        // NOTE: This getSize code is copied from DynamicDrawableSpan and modified
+        // to not use a Drawable
+
+        fm?.let {
+            it.ascent = -_height
+            it.descent = 0
+
+            it.top = it.ascent
+            it.bottom = 0
+        }
+
+        return _width
+    }
+
+    public override fun setTextView(textView: TextView?) {
+        this.textView = textView
+    }
+
+    public override fun draw(
+        canvas: Canvas,
+        text: CharSequence,
+        start: Int,
+        end: Int,
+        x: Float,
+        top: Int,
+        y: Int,
+        bottom: Int,
+        paint: Paint
+    ) {
+        if (drawable == null) {
+            val imageRequestBuilder = ImageRequestBuilder.newBuilderWithSource(_uri)
+            val imageRequest: ImageRequest =
+                ReactNetworkImageRequest.fromBuilderWithHeaders(imageRequestBuilder, headers)
+
+            draweeHolder.hierarchy.setActualImageScaleType(ImageResizeMode.toScaleType(resizeMode))
+
+            draweeControllerBuilder.reset()
+            draweeControllerBuilder.oldController = draweeHolder.controller
+
+            callerContext?.let { draweeControllerBuilder.setCallerContext(it) }
+
+            draweeControllerBuilder.setImageRequest(imageRequest)
+
+            val draweeController = draweeControllerBuilder.build()
+            draweeHolder.controller = draweeController
+            draweeControllerBuilder.reset()
+
+            checkNotNull(draweeHolder.topLevelDrawable).let { _drawable ->
+                _drawable.setBounds(0, 0, _width, _height)
+                if (tintColor != 0) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                        _drawable.colorFilter = BlendModeColorFilter(tintColor, BlendMode.SRC_IN)
+                    } else {
+                        _drawable.setColorFilter(tintColor, PorterDuff.Mode.SRC_IN)
+                    }
+                }
+                _drawable.callback = textView
+                drawable = _drawable
+            }
```

Comment:
Looks way cleaner now! if you want to improve it even further, you can avoid using `_drawable` as `apply` scopes the variable into the block. Something like this:

```suggestion
          checkNotNull(draweeHolder.topLevelDrawable).apply {
            setBounds(0, 0, _width, _height)
            if (tintColor != 0) {
                colorFilter = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                    BlendModeColorFilter(tintColor, BlendMode.SRC_IN)
                } else {
                   setColorFilter(tintColor, PorterDuff.Mode.SRC_IN)
                }
            }
            callback = textView
            drawable = this
        }
```

Reply:
I had tried that implementation haha. Should've kept it:

https://github.com/facebook/react-native/blob/16d992c6140bacf90db9dfdef4cf7afa7f3ffa92/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/frescosupport/FrescoBasedReactTextInlineImageSpan.kt#L145-L156

let me revert to that.

Reply:
done